nixnet/_funcs.py

```diff
@@ -695,3 +695,43 @@ def nxdb_undeploy(
         database_alias_ctypes,
     )
     _errors.check_for_error(result.value)
+
+
+def nxdb_get_database_list(
+    ip_address,  # type: typing.Text
+):
+    # type: (...) -> typing.Tuple[List[typing.Text], List[typing.Text], int]
+    ip_address_ctypes = _ctypedefs.char_p(ip_address.encode('ascii'))
+    size_of_alias_buffer, size_of_filepath_buffer = nxdb_get_database_list_sizes(ip_address)
+    size_of_alias_buffer_ctypes = _ctypedefs.u32(size_of_alias_buffer)
+    size_of_filepath_buffer_ctypes = _ctypedefs.u32(size_of_filepath_buffer)
+    alias_buffer_ctypes = ctypes.create_string_buffer(size_of_alias_buffer)
+    filepath_buffer_ctypes = ctypes.create_string_buffer(size_of_filepath_buffer)
+    number_of_databases_ctypes = _ctypedefs.u32()
+    result = _cfuncs.lib.nxdb_get_database_list(
+        ip_address_ctypes,
+        size_of_alias_buffer_ctypes,
+        alias_buffer_ctypes,
+        size_of_filepath_buffer_ctypes,
+        filepath_buffer_ctypes,
+        number_of_databases_ctypes,
+    )
+    _errors.check_for_error(result.value)
+    alias_list = alias_buffer_ctypes.value.decode("ascii").split(",")
+    filepath_list = filepath_buffer_ctypes.value.decode("ascii").split(",")
+    return list(zip(alias_list, filepath_list))
```

Comment:
I left the decoding of the strings logic in this layer thinking that this would need to be generated for all string output types anyway.

I've moved  the 'split' and 'zip' logic to the layer above (databases) though.